### DISCUSSION TITLE
dispose type :  "LineSegments"

### DISCUSSION THREAD

**Discussion by arpu** [donmccurdy/glTF-Transform#769] [Q&A] [answered]
Hello,

trying to dispose all LineSegments in a glb model , any hint how i should list and dispose this?

**Comment by donmccurdy** [comment, marked as answer]
Hi! If you expect the model to contain only triangle meshes, then a script like this would clean anything else up:


```javascript
import { Primitive } from '@gltf-transform/core';
import { prune } from '@gltf-transform/functions';

for (const mesh of document.getRoot().listMeshes()) {
  for (const prim of mesh.listPrimitives()) {
    if (prim.getMode() !== Primitive.Mode.TRIANGLES) {
      prim.dispose();
    }
  }

  if (mesh.listPrimitives().length === 0) {
    mesh.dispose();
  }
}

await document.transform(prune());
```

Full list of primitive modes here:

https://github.com/donmccurdy/glTF-Transform/blob/cc0103d7e447853d24250e012e77c58cf45688ec/packages/core/src/properties/primitive.ts#L58-L80

**Reply by arpu** [reply]
Awesome, works perfect!
